test/test.js

```diff
@@ -376,6 +376,36 @@ suite('Response', function() {
       return json;
     })
   })
+  test('clone text response', function() {
+    var r = new Response('{"foo":"bar"}', {headers: {'content-type': 'application/json'}}),
+        clone = r.clone();
+    assert(clone.headers !== r.headers, true);
+    assert(clone.headers.get('content-type') === r.headers.get('content-type'), true);
+    return Promise.all([clone.json(), r.json()]).then(function(jsons){
+      assert(jsons[0].foo, 'bar');
```

Comment:
Thanks, I'm more used to expect(...).to... syntax. Give me a sec, I'll upload update.